contracts/Gateway.sol

```diff
@@ -128,16 +136,19 @@ contract Gateway is ProtocolVersioned, Owned {
         // check if the stake request does not exists
         require(stakeRequests[msg.sender].beneficiary == address(0));
 
-        require(OpenSTValueInterface(openSTProtocol).valueToken().transferFrom(msg.sender, address(this), _amount));
+        uint256 amountUT = TokenConversionLib.calculateUTAmount(_amount, conversionRate, conversionRateDecimals);
+        uint256 stakedAmount = TokenConversionLib.calculateSTAmount(amountUT, conversionRate, conversionRateDecimals);
```

Comment:
See comments on the new lib functions, but I don't think we need to calculate amountUT, do we? We just need to pass the `_amount` and get in return `stakedAmount`; since `amountUT` is obtained from the lib and then immediately sent back to the lib, it seems like that can all be done in one call rather than two.

*Alternatively* (but this might be a better question for refactoring), does it make sense to leave the two calls as is, but also store `amountUT` in the `stakeRequest` and pass it in the call to `stake` (which would remove the need to call `calculateUTAmount` in `OpenSTValue`)?

Reply:
1. Are you suggesting to create the third method in the library which internally calculates the stake amount(using these two methods)?  These two methods are required, as they are used in OpenST-Value as well.  



2. The second point about refactoring can be taken care with 0.9.4 Gateway development. Please let me know if you have another opinion.